### PR TITLE
Upgrade deConz to v2.28.1

### PR DESCRIPTION
**Pull request by jannickfahlbusch**
This upgrades deConz from 2.25.3 to [2.27.6](https://github.com/dresden-elektronik/deconz-rest-plugin/releases/tag/v2.27.6).

I've built the addon locally but had to [ignore kernel modules in the config](https://github.com/home-assistant/addons/blob/100eef68299de67daf7c4cfca0b68ded46ad786c/deconz/config.yaml#L26) as it failed otherwise in a local docker based dev environment. I'm not sure if this is due to the dev-environment or due to an issue with the addon.

Apart from that I am able to access the deConz VNC UI as well as the Phoscon UI:
image

I was able to control a few IKEA switches and receive data from my Aqara temperature sensors. 

## Summary by CodeRabbit

- **New Features**
	- Upgraded to version 7.0.0, introducing significant updates and improvements.
	- DECONZ software version upgraded to 2.28.1, enhancing performance and stability.
	- Transitioned to Debian Bookworm for all architecture types, improving compatibility.
	- Changed window manager from wmii to dwm, potentially enhancing user interface experience.

- **Bug Fixes**
	- Upgraded to version 2.28.1 of DECONZ, which includes various bug fixes.

- **Documentation**
	- Updated changelog to reflect the changes and version updates accurately.

### CODE REVIEW

File: deconz/build.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ codenotary:
   signer: notary@home-assistant.io
   base_image: notary@home-assistant.io
 args:
-  DECONZ_VERSION: 2.25.3
+  DECONZ_VERSION: 2.26.7
```
Comment (hiteule) [inline deconz/build.yaml:10]: Here is a typo. It should be 2.27.6 I guess.
```suggestion
  DECONZ_VERSION: 2.27.6
```

Comment (frenck) [reply deconz/build.yaml:10]: Interesting, show how was this PR tested?

Comment (jannickfahlbusch) [reply deconz/build.yaml:10]: I committed from the wrong folder 🙈 
During local testing I copied the `deconz` addon folder to a local one to only have this addon for testing as I tried following [the documentation about local testing](https://developers.home-assistant.io/docs/add-ons/testing/). 
In order to commit the changes I should have synced the changes back but did them manually...

Comment (jannickfahlbusch) [reply deconz/build.yaml]: I just checked the version in my dev-installation and can confirm that I indeed tested the 2.27.6 version.

The docker image also shows deCONZ in the desired version:
```bash
docker run -it --entrypoint bash local/homeassistant/aarch64-addon-deconz:6.24.0
root@a07e9d4a3358:/# dpkg -l deconz
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  deconz         2.27.6       arm64        Zigbee monitoring and control.
```

**Comment by jannickfahlbusch** [comment]
Somehow the image build for `armhf` is failing to compile WiringPi:
```
#0 198.5 + git clone --depth 1 https://github.com/WiringPi/WiringPi /usr/src/wiringpi
#0 198.5 Cloning into '/usr/src/wiringpi'...
#0 199.7 + cd /usr/src/wiringpi/wiringPi
#0 199.7 + make
#0 199.8 [Compile] wiringPi.c
#0 200.1 wiringPi.c: In function ‘pullUpDnControlDevice’:
#0 200.1 wiringPi.c:1817:28: error: ‘GPIOHANDLE_REQUEST_BIAS_DISABLE’ undeclared (first use in this function); did you mean ‘GPIOHANDLE_REQUEST_OPEN_DRAIN’?
#0 200.1    unsigned int biasflags = GPIOHANDLE_REQUEST_BIAS_DISABLE | GPIOHANDLE_REQUEST_BIAS_PULL_UP | GPIOHANDLE_REQUEST_BIAS_PULL_DOWN;
#0 200.1                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#0 200.1                             GPIOHANDLE_REQUEST_OPEN_DRAIN
#0 200.1 wiringPi.c:1817:28: note: each undeclared identifier is reported only once for each function it appears in
#0 200.1 wiringPi.c:1817:62: error: ‘GPIOHANDLE_REQUEST_BIAS_PULL_UP’ undeclared (first use in this function); did you mean ‘GPIOHANDLE_REQUEST_INPUT’?
#0 200.1    unsigned int biasflags = GPIOHANDLE_REQUEST_BIAS_DISABLE | GPIOHANDLE_REQUEST_BIAS_PULL_UP | GPIOHANDLE_REQUEST_BIAS_PULL_DOWN;
#0 200.1                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#0 200.1                                                               GPIOHANDLE_REQUEST_INPUT
#0 200.1 wiringPi.c:1817:96: error: ‘GPIOHANDLE_REQUEST_BIAS_PULL_DOWN’ undeclared (first use in this function); did you mean ‘GPIOHANDLE_REQUEST_ACTIVE_LOW’?
#0 200.1    unsigned int biasflags = GPIOHANDLE_REQUEST_BIAS_DISABLE | GPIOHANDLE_REQUEST_BIAS_PULL_UP | GPIOHANDLE_REQUEST_BIAS_PULL_DOWN;
#0 200.1                                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#0 200.1                                                                                                 GPIOHANDLE_REQUEST_ACTIVE_LOW
#0 200.3 make: *** [Makefile:83: wiringPi.o] Error 1
```

**Comment by jannickfahlbusch** [comment]
I've upgraded the base image from Debian buster to bookworm as well.

The deCONZ VNC session looks now slightly different due to the change from wmii (Which is not available anymore; I was also unable to compile it myself due to missing libraries) as window manager to dwm:
image

The Phoscon UI works as well:
image

I don't have a ConBee II right now, I hope to find the time for testing this evening.

**Comment by jannickfahlbusch** [comment]
Just installed the new version in my dev instance. Everything is working fine, the ConBee II is being detected and can be used to interact with my ZigBee network.

**Comment by woytekbode** [comment]
@frenck Apologies beforehand, don't want to push! However, considering the current Deconz addon version has some serious issues since early this year and the previous version was skipped due to other issues, it would be really nice to get this latest release out there. Any chance someone could do a review on this PR?

**Comment by marmarek** [comment]
FWIW I'm using this PR and it works fine for me.

**Comment by g1za** [comment]
Maybe a review of this PR is also something @Kane610 can help with?

**Comment by Mimiix** [comment]
Hi @frenck , what's holding the merge?

**Comment by Kane610** [comment]
> Maybe a review of this PR is also something @Kane610 can help with?

Sorry I am not involved at all with the addon

**Comment by yan14** [comment]
@frenck I don't get it. My PR (https://github.com/home-assistant/addons/pull/3702) was closed after a couple of days, because I couldn't test it. This one is tested by several users and it doesn't get a review for weeks!

**Comment by Mimiix** [comment]
> > @frenck I don't get it. My PR (#3702) was closed after a couple of days, because I couldn't test it. This one is tested by several users and it doesn't get a review for weeks!
> 
> Don't blame frenck - very likely busy doing other HA (Core) related things, maintaining addons seems not to be number one on the todo list I guess.
> 
> Dresden should take over control and maintenance of the official deCONZ addon for HA. Otherwise this situation will NEVER EVER improve. HA has its own ZigBee solution (ZHA) and basically absolutely no interest in maintaining a competitor's addon. The math in my opinion is:
> 
> `Conflict of interests from HA side (just a wild guess)` + `lack of resources from Dresden side` = **`completely disappointing experience for deCONZ HA users`**
> 
> This comment will 100 % be minimized/marked as off-topic, but things will escalate and users start to be reaaaaaally upset month by month this situation continues.

I guess @manup would be happy to be a reviewer and approve the pr if he got permissions to do so :)

**Comment by emlnpro** [comment]
> > > @frenck I don't get it. My PR (#3702) was closed after a couple of days, because I couldn't test it. This one is tested by several users and it doesn't get a review for weeks!
> > 
> > 
> > Don't blame frenck - very likely busy doing other HA (Core) related things, maintaining addons seems not to be number one on the todo list I guess.
> > Dresden should take over control and maintenance of the official deCONZ addon for HA. Otherwise this situation will NEVER EVER improve. HA has its own ZigBee solution (ZHA) and basically absolutely no interest in maintaining a competitor's addon. The math in my opinion is:
> > `Conflict of interests from HA side (just a wild guess)` + `lack of resources from Dresden side` = **`completely disappointing experience for deCONZ HA users`**
> > This comment will 100 % be minimized/marked as off-topic, but things will escalate and users start to be reaaaaaally upset month by month this situation continues.
> 
> I guess @manup would be happy to be a reviewer and approve / merge the pr if he got permissions to do so :)

Yes as user I can't disagree with that, terrible experience waiting month for deconz addon update in HA.. 
Maybe a unofficial module up to date can be done aside ?

**Comment by frenck** [comment]
Thanks for all the pings, but I am traveling with my family for a few weeks (greetings from the top of the mountains in Austria 👋)

/CC @agners can you pick up?

**Comment by manup** [comment]
Hey everyone, it would be really cool if we can figure out how to improve the release/merge time slots.

I'm open to help maintaining and push releases for the add-on. But have to admit that my knowledge of the inner workings of HA are very limited. Pushing updates of deCONZ is rather simple, but if HA <-> add-on interaction needs to be updated I'd need a helping hand.

Frenck hope you have a great time in Austria and time to relax. Perhaps we can schedule a discussion/meeting afterwards to figure out how to move forward.

**Comment by dutch-erik** [comment]
/CC @agners 
There is a new deCONZ version -> [2.28.1](https://github.com/dresden-elektronik/deconz-rest-plugin/releases/tag/v2.28.1)
So hopefully will this addon be upgraded soon now?

**Comment by jannickfahlbusch** [comment]
I’ll update the PR this evening and will try the new version in a HA test instance

**Comment by dusopn** [comment]
wow, that would be so cool if you could, we on HA addon are waiting way too long since last update, especially considering so many cool deconz updates since then... thanks for your time!

**Comment by jannickfahlbusch** [comment]
> I'm open to help maintaining and push releases for the add-on. [...] Perhaps we can schedule a discussion/meeting afterwards to figure out how to move forward.

@manup would this be something you'd do in your spare time or "officially" from Dresden Elektronik? 
If you would need help for the unofficial support of the HA addon feel free to invite me as well.

**Comment by jannickfahlbusch** [comment]
image

Building and starting the new version works. However it seems like my spare Conbee broke, it is not recognized anymore as an USB device :/

During building I still had to disable the usage of kernel modules, otherwise I am unable to start the addon on a n M2 Mac.
I'll investigate this further on a Linux box - we should wait with the merge until I was able to start deCONZ with kernel modules enabled.

**Comment by Mimiix** [comment]
> > I'm open to help maintaining and push releases for the add-on. [...] Perhaps we can schedule a discussion/meeting afterwards to figure out how to move forward.
> 
> @manup would this be something you'd do in your spare time or "officially" from Dresden Elektronik? If you would need help for the unofficial support of the HA addon feel free to invite me as well.

Manup is the product owner of deCONZ.

**Comment by Mimiix** [comment]
> image
> Building and starting the new version works. However it seems like my spare Conbee broke, it is not recognized anymore as an USB device :/
> 
> During building I still had to disable the usage of kernel modules, otherwise I am unable to start the addon on a n M2 Mac. I'll investigate this further on a Linux box - we should wait with the merge until I was able to start deCONZ with kernel modules enabled.

Why not just merge the proved working 2.27.x version and do this one after?

**Comment by jannickfahlbusch** [comment]
We can also do that, we would need somebody to finally review and merge this though.

**Comment by bcutter** [comment]
Looking at the update interval track history I'd vote for updating to the very last available stable version (who knows how long it would take for the next addon release) instead of starting right away with an already superseded stable version (2.27.1). Waiting a few more days to check what's the problem with 2.28.1 (couldn't spot any fundamental obvious changes in the release notes) seems to be the overall better option. /2 cents

**Comment by dutch-erik** [comment]
So we are again waiting (after months of waiting) because 1 person has an issue?
What else is needed to push the HA addon?

**Comment by jannickfahlbusch** [comment]
I now successfully tested v2.28.1 on my productive HA instance.

image
image

@agners could you please take a look here?

File: deconz/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 6.24.0
```
Comment (agners) [inline deconz/CHANGELOG.md:3]: Maybe we should bump the version to 7.0.0 here? This seems a rather bigger update which warrants a major version bump. The 6.0.0 bump was also done when moving to a new Debian version.

**Comment by PolestarWx** [comment]
> I now successfully tested v2.28.1 on my productive HA instance.
> 
> image image
> @agners could you please take a look here?

Have you vérified that add device page works under deconz ? 
Because in another fork I had problem

**Comment by jannickfahlbusch** [comment]
> Have you vérified that add device page works under deconz ?
Because in another fork I had problem but could be because of my reverse proxy conf

If you mean within the Phoscon UI then yes:
image
image

**Comment by agners** [comment]
> Have you vérified that add device page works under deconz ?

Adding a device via Phoscon worked here too.

I do get this in the logs constantly, this wasn't the case before :thinking: 

```
[17:38:47] INFO: Running the deCONZ OTA updater...
[17:38:48] INFO: Running the deCONZ OTA updater...
```

**Comment by jannickfahlbusch** [comment]
Probably because the file is not existing anymore. Where did this come from in the first place?

[Searching for the file on GitHub returns no results](https://github.com/search?q=path%3A*%2FdeCONZ-otau-dl.sh&type=code), when searching for the string I only get occurrences in [forks of this repository](https://github.com/search?q=deCONZ-otau-dl.sh&type=code).

**Comment by agners** [comment]
> As the OTA location from the script does not exist anymore I'd consider the removal of the service to be safe.

Yeah seems the right thing to do then even.

I am not familiar with deCONZ OTA, maybe this is built-in now. If something needs to be done to the add-on, we can readd the feature in a follow up PR.

**Comment by PolestarWx** [comment]
Wowww, thanks you so much guy, with this version we can upload our definition file it's a game changer 
Thank you after all this time waiting ❤️

**Comment by Mimiix** [comment]
> > As the OTA location from the script does not exist anymore I'd consider the removal of the service to be safe.
> 
> Yeah seems the right thing to do then even.
> 
> I am not familiar with deCONZ OTA, maybe this is built-in now. If something needs to be done to the add-on, we can readd the feature in a follow up PR.

Great that you were able to merge.

I'd like to invite you as a developer on our deconz discord to make this much smoother in the future. Dresden Elektronik is there as well :)

**Comment by MEKadan** [comment]
I do not is this a correct place for this, but I don't know others either...

My deconz UI looks like this (HAOS on RPi4, all latest updates)

image

How do I get rid of that grey panel ("VNC Config") on the right side of the screen?
Or at least resize it significantly smaller?

There are no handles or anything like in the panel. None of the menu items seem to have any affect on it either.

**Comment by PolestarWx** [comment]
> I do not know is this a correct place for this, but I don't know others either...
> 
> My deconz UI looks like this (HAOS on RPi4, all latest updates)
> 
> image
> 
> How do I get rid of that grey panel ("VNC Config") on the right side of the screen? Or at least resize it significantly smaller?
> 
> There are no handles or anything like in the panel. None of the menu items seem to have any affect on it either.

Not the right place at all, but in update change there is a new window manger, with new keyboard command to close windows, no more mouse unfortunately

**Comment by Mimiix** [comment]
This has nothing to do with deconz, this looks like it's from the vnc cliënt in use.

**Comment by MEKadan** [comment]
> This has nothing to do with deconz, this looks like it's from the vnc cliënt in use.

Yup, I updated Home Assistant deconz addon 6.2.3 to 7.00 and this is how it looks now in it's VNC client. Is there a thread for it? I could not find.

**Comment by emlnpro** [comment]
You can close it with dwm window manager command ;)

**Comment by MEKadan** [comment]
> You can close it with dwm window manager keyboard shortcut ;) alt - shift - C when window is active

OK, thanks. I did not realize that it might work with keyboard shortcuts, when previously everything was doable with mouse...

**Comment by dusopn** [comment]
Thank you for the update guys!

**Comment by prof7bit** [comment]
May I suggest to finally switch to a PROPER window manager that can be used with the mouse like virtually EVERY other window manager out there since the invention of the mouse instead of forcing one of these exotic experimental incomplete unusable toy window manager experiments onto all the innocent users?

**Comment by bcutter** [comment]
> May I suggest to finally switch to a PROPER window manager that can be used with the mouse like virtually EVERY other window manager out there since the invention of the mouse instead of forcing one of these exotic experimental incomplete unusable toy window manager experiments onto all the innocent users?

Let's keep this PR focused. For Window manager issues you can turn to https://forum.phoscon.de/t/new-deconz-gui-windows-control-dwm-issues/5097.

**Comment by bcutter** [comment]
> * Changed window manager from wmii to dwm, potentially enhancing user interface experience.

@jannickfahlbusch Is this an upstream change, so part of deCONZ? Because that "experience" was not really enhanced, instead the new DWM creates some serious issues, making deCONZ almost unusable and also leading to the addon being restarted (maybe by the watchdog or some deCONZ internal kill switch).

**Comment by jannickfahlbusch** [comment]
No this was done as the previously used wmii is not compileable anymore. The libraries for it are missing and the documentation is partially gone. 
Dwm appears to come from the same authors and was named as successor of wmii so I adapted to it. 

Not sure why wmii was chosen in the first place, with this change I didn’t intend to switch from a tiling wm to a different paradigm. 

In hindsight we probably should have had a discussion about the window manager used for the addon here. Sorry for the trouble. 

Ideally deConz would offer a web UI or Phoscon would receive all of the features that deConz offers. For now it would probably be the best thing to adapt the window manager to something more user friendly. 
The resource usage of the addon would increase with that so there might be other users that are unable to run the addon due to the increased resource utilization though.

**Comment by bcutter** [comment]
> In hindsight we probably should have had a discussion about the window manager used for the addon here. Sorry for the trouble.

Probably - but that's what "normal users" should expect from the ones who maintain an addon. I might be wrong but you luckily stepped in to allow all us HA users to finally catch up with the 6 months old deCONZ releases and you're neither affiliated with Dresden Elektronik nor Nabu Casa/HA team, so you gave your best to make us happy :-) Thank you for that once again. Short: nothing to apologize for.

**The question is, how to solve those issues** (outlined in last few posts here or at https://forum.phoscon.de/t/new-deconz-gui-windows-control-dwm-issues/5097)**?**

> Ideally deConz would offer a web UI or Phoscon would receive all of the features that deConz offers.

I don't think this is a realistic hope, at least not on short-term. But who knows what Dresden Elektronik is working on and what's on the strategic map when it comes to deCONZ vs. Phoscon and a potential timeline?

There are few smaller workarounds from Dresden team (e. g. for the DDF editor at https://github.com/dresden-elektronik/deconz-rest-plugin/pull/7910), but no overall solution to handle all the other issues which came up with the new window manager like
- how to easily close/minimize the grey default VNC window when the addon starts (looks like the same trick as in https://github.com/dresden-elektronik/deconz-rest-plugin/pull/7910 is not possible here?)
- how to reposition or maximize windows
- how to minimize windows like we could in the past when they got vertically aligned, not horizontally stacked as they are now
- avoid usage of the [ALT] + [SHIFT] + [C] shortcut which leads to addon restart once applied to the main window (which sometimes is necessary to close because otherwise the view is stuck with a 50 % black screen and a 50 % main window unable to control)

TBH I have no idea what an actual solution could look like. But I'm sure we'll (urgently) need one as more and more users now updating will run into those issues.

_Resource usage by the way currently is not an issue, CPU and MEM of the addon still very calm, no change to previous addon versions (relative data provided by supervisor integration).
grafik_

**Comment by agners** [comment]
> In hindsight we probably should have had a discussion about the window manager used for the addon here. Sorry for the trouble.

As @bcutter said, don't feel bad, you did nothing wrong here: You've chosen the closest replacement. Now it turns out that this is problematic, but it seems that also the previous version wasn't liked. Don't get discouraged by negative comments. Sure, there is probably a problem, but you actually put work in to this, and most folks are thankful for that. Let's just go from here.

> The resource usage of the addon would increase with that so there might be other users that are unable to run the addon due to the increased resource utilization though.

The add-on is 700MB anyways, I don't think that an alternative DE would be that much different in terms of resources. Probably worth looking at is LXDE as it is known to be user friendly and lightweight.

I guess one of the main features required is VNC capability. Recently wayvnc got quite popular in the Wayland space, so Wayland compositors might be an option. Weston (the Wayland reference compositor) has a VNC backend, or wlroots has a headless backend too and supports VNC afaik.

**Comment by wtaferner** [comment]
For what it is worth, here a link to a handy [DWM Cheatsheet](https://gist.github.com/erlendaakre/12eb90eef84a3ab81f7b531e516c9594).

[Mod] = [L ALT]

I switched now to a remote VNC client as on my RPi the embedded VNC is slow for obvious reasons and paired with the cheatsheet the handling is subjectively better than before, but I was never used to the old GUI anyway.

**Comment by workoutpeter** [comment]
Is it possible to change the websocket port? Openthread border router use 8081 too. Any changes in config does'nt work.

**Pull request merged**